framework/src/media/audio/audio_manager.c

```diff
@@ -321,6 +355,44 @@ static audio_manager_result_t get_supported_sample_rate(int fd, uint32_t *sample
 	return AUDIO_MANAGER_SUCCESS;
 }
 
+static audio_manager_result_t audio_manager_mutex_lock(audio_card_type_t type)
+{
+	if (type == INPUT) {
+		if (g_actual_audio_in_card_id < 0) {
```

Comment:
As I mentioned line 358, this checking is not neccessary

Reply:
we can use pthread_mutex_lock directly.

Reply:
Modified.